python/core/qgsstringstatisticalsummary.sip

```diff
@@ -1,165 +1,220 @@
+ variants to a list and calling calculateFromVariants().
+ \param value variant to add
+.. note::
+
+   call reset() before adding the first string using this method
+ to clear the results from any previous calculations
+.. note::
+
+   finalize() must be called after adding the final value and before
+ retrieving calculated statistics.
+.. seealso:: calculateFromVariants()
+.. seealso:: finalize()
+%End
 
-    /** Calculates summary statistics for an entire list of strings at once.
-     * @param values list of strings
-     * @see calculateFromVariants()
-     * @see addString()
-     */
-    void calculate( const QStringList& values );
-
-    /** Calculates summary statistics for an entire list of variants at once. Any
-     * non-string variants will be ignored.
-     * @param values list of variants
-     * @see calculate()
-     * @see addValue()
-     */
-    void calculateFromVariants( const QVariantList& values );
-
-    /** Adds a single string to the statistics calculation. Calling this method
-     * allows strings to be added to the calculation one at a time. For large
-     * quantities of strings this may be more efficient then first adding all the
-     * strings to a list and calling calculate().
-     * @param string string to add
-     * @note call reset() before adding the first string using this method
-     * to clear the results from any previous calculations
-     * @note finalize() must be called after adding the final string and before
-     * retrieving calculated statistics.
-     * @see calculate()
-     * @see addValue()
-     * @see finalize()
-     */
-    void addString( const QString& string );
-
-    /** Adds a single variant to the statistics calculation. Calling this method
-     * allows variants to be added to the calculation one at a time. For large
-     * quantities of variants this may be more efficient then first adding all the
-     * variants to a list and calling calculateFromVariants().
-     * @param value variant to add
-     * @note call reset() before adding the first string using this method
-     * to clear the results from any previous calculations
-     * @note finalize() must be called after adding the final value and before
-     * retrieving calculated statistics.
-     * @see calculateFromVariants()
-     * @see finalize()
-     */
-    void addValue( const QVariant& value );
-
-    /** Must be called after adding all strings with addString() and before retrieving
-     * any calculated string statistics.
-     * @see addString()
-     */
     void finalize();
+%Docstring
+ Must be called after adding all strings with addString() and before retrieving
+ any calculated string statistics.
+.. seealso:: addString()
+%End
 
-    /** Returns the value of a specified statistic
-     * @param stat statistic to return
-     * @returns calculated value of statistic
-     */
     QVariant statistic( Statistic stat ) const;
```

Comment:
you might need to add `QgsStatisticalSummary`in front of `Statistic` (in the header)